app/src/main/java/org/oppia/android/app/profileprogress/ProfileProgressFragmentPresenter.kt

```diff
@@ -70,8 +70,8 @@ class ProfileProgressFragmentPresenter @Inject constructor(
   }
 
   private fun createRecyclerViewAdapter(): BindableAdapter<ProfileProgressItemViewModel> {
-    return BindableAdapter.MultiTypeBuilder
-      .newBuilder<ProfileProgressItemViewModel, ViewType> { viewModel ->
+    return BindableAdapter.MultiTypeBuilder.Factory(fragment)
```

Comment:
Definitely in the right direction, except we should be injecting the factory rather than constructing it (similar to ``PersistentCacheStore``). The constructor declaration below that's taking in fragment is correct.

Reply:
Done.